src/ray/object_manager/object_manager.cc

```diff
@@ -278,6 +278,7 @@ void ObjectManager::PullEstablishConnection(const ObjectID &object_id,
 
   if (conn != nullptr) {
     PullSendRequest(object_id, conn);
+    connection_pool_.ReleaseSender(ConnectionPool::ConnectionType::MESSAGE, conn);
```

Comment:
It seems like we’ll have to open and remove a new connection every time we request an object from a remote manager, is that right? Is that a performance issue?

Reply:
This doesn't remove it right, just releases it back into the pool.

Reply:
Ah right, @stephanie-wang, the *key fix* is essentially moving this `ReleaseSender` call from the `AsyncWriteMessage` callback to here, right?

Reply:
Yes.

Reply:
Ok, I see that you answered this in the PR description.

Reply:
@robertnishihara Thanks for pointing it out.